src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -1758,9 +1759,48 @@ def discriminant(self):
 
         return (MatrixSpace(QQ, 4, 4)(L)).determinant().sqrt()
 
+    def is_maximal(self):
+        r"""
+        Return ``True`` if the order is maximal in the ambient quaternion algebra, ``False`` otherwise
+        Only works in quaternion algebras over number fields
+
+        OUTPUT: Boolean
+
+        EXAMPLES::
+
+            sage: p = 11
+            sage: B = QuaternionAlgebra(QQ,-1,-p)
+            sage: i,j,k = B.gens()
+            sage: O0_basis = (1,i,(i+j)/2, (1+i*j)/2)
+            sage: O0 = B.quaternion_order(O0_basis)
+            sage: O0.is_maximal()
+            True
+            sage: O1 = B.quaternion_order([1,i,j,i*j])
+            sage: O1.is_maximal()
+            False
+
+        TESTS::
+
+            sage: B = QuaternionAlgebra(GF(13),-1,-11)
+            sage: i,j,k = B.gens()
+            sage: O0_basis = (1,i,j, k)
+            sage: O0 = B.quaternion_order(O0_basis)
+            sage: try:
+            ....:     O0.is_maximal()
+            ....:     print("This should fail")
+            ....: except NotImplementedError:
+            ....:     print("ok")
+            ....:
+            ok
+        """
+        if not(self.quaternion_algebra().base_ring() in NumberFields()):
+            raise NotImplementedError("Check for maximality is only implemented for quaternion algebras over number fields).")
+        return(self.discriminant() == self.quaternion_algebra().discriminant())
+
     def left_ideal(self, gens, check=True, *, is_basis=False):
         r"""
         Return the left ideal of this order generated by the given generators.
+        Only works in quaternion algebras over number fields.
```

Comment:
Please add an empty line after the first sentence of the docstring (see dcoudert's comment above).